### ISSUE TITLE
au run fails when the app wasn't generated by that PC.

### ISSUE THREAD

**Issue by jaeko44** [aurelia/cli#253]
**I'm submitting a bug report**

* **Library Version:**
major

**Please tell us about your environment:**
* **Operating System:**
Windows [10]

* **Node Version:**
4.4.7

* **NPM Version:**
3.8.9

* **Language:**
ESNext

**Current behavior:**
When uploading a repo to GitHub, I ignore node_modules as usual.
When pulling the project, followed by:
`npm install && au run`
It will lead with an error and won't run. The only way I found to get the app to work is by using `au new` to generate a new project and then replacing the SRC files...

* **What is the expected behavior?**
To run

* **What is the motivation / use case for changing the behavior?**
logic

**Comment by TylerJPresley** [comment]
I just had this issue.

**Comment by EisenbergEffect** [comment]
What error?

**Comment by EisenbergEffect** [comment]
You would need to have the cli installed on that machine along with all pre-requisities, paticuarly npm 3

**Comment by TylerJPresley** [comment]
Example. I started this at work to show Charles what I was seeing. This was created on a windows machine. https://github.com/TylerJPresley/au-layout

Later when I pulled it down on my MacBook at home it could do anything with it. It would not bundle and just kept throwing errors. I had to create a new project and copy my src folder over to get it to run on my MBP.

**Comment by TylerJPresley** [comment]
Work PC - NPM: 3.10.6 / Windows 10
Home PC - NPM: 3.10.6 / OS X 10.11.6

**Comment by EisenbergEffect** [comment]
We really need error info. We haven't been able to reproduce this.

**Comment by TylerJPresley** [comment]
Give me a bit

**Comment by TylerJPresley** [comment]
Pull this project down on you computer (Step 4). 

1. On my windows machine I created a new project using aurelia-cli 0.17.0.
   a. `au new` -> name: `au-test` -> `1` -> `enter` -> `enter`

2. Run the app with `au run --env dev --watch`
   a. App works. I can see welcome at `http://localhost:9000`

3. Uploaded to github

4. Cloned on mac
   a. `git clone git@github.com:TylerJPresley/au-test.git`
   b. `npm install`
   c. `au run --env dev --watch`
   d. opened `http://localhost:9000`

5. ERROR: 
   a. main.js:1 Uncaught SyntaxError: Unexpected token import
   b. vendor-bundle.js:11052 Uncaught TypeError: Cannot read property '__useDefault' of undefined

Link to that line: https://github.com/TylerJPresley/au-test/blob/master/scripts/vendor-bundle.js#L11052

**Comment by TylerJPresley** [comment]
I was able to duplicate it in this test. This is a completely stock/brand new project. None of it was modified. Just the commands above.

**Comment by jaeko44** [comment]
Also reproducing here, give me a moment.

**Comment by TylerJPresley** [comment]
Something to note. When I delete app-bundle, app-bundle map, and vendor bundle.. If I run au run again the app-bundle file is blank and the map file is not generated on the 2nd computer.

**Comment by jaeko44** [comment]
Things noted:

1. It doesn't need to be a different machine for it to fail a cloned deployment.
2. The node_modules folder is much larger in the failed deployment. 
     - Working proj DIR node_modules size = 90.1MB
     - Failed delpoyment DIR node_modules size = 264MB
3. The error - ![Error](http://puu.sh/ql3Qc/00064bfc9b.png )
4. Trying to manually install 'aurelia-binding' leads to additional dependencies that are in the aurelia_project/aurelia.json dependency list. 
5. Installing all the dependencies using --save will lead to an error (as I tried earlier, don't have the error now unless I try again).
6. Running 'au new' and then copying the node_modules folder from the new generated application and replacing it with the failed deployment will fix the project and it can then be run again.

**Comment by TylerJPresley** [comment]
So the dependencies installed by the CLI are different from the ones installed by mom install?

**Comment by zamotany** [comment]
Could not been able to reproduce on the same machine.

**Environment**

-   **OS:**  
    Windows 10

-   **Node:**  
    6.2.2

-   **NPM:**  
    3.9.5

-   **Language:**  
    Typescript

  
Steps:

1.  Cloned.
2.  `npm install`
3.  `npm run typings install` (in `package.json` I have `“scripts”: { “typings”: “typings” }`
4.  `au run`

**Comment by jaeko44** [comment]
I've been able to reproduce @TylerJPresley error on a linux environment (Ubuntu 14.04)

**Please tell us about your environment:**
* **Operating System:**
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.4 LTS

* **Node Version:**
v6.3.1

* **NPM Version:**
3.10.3

* **Language:**
ES6 (au-test)

* **Steps:**

1. git clone https://github.com/TylerJPresley/au-test.git
2. cd au-test
3. npm install
4. au run

* **Result:**

No error when running 'au run', however the app fails to load via browser.

`main.js:1 Uncaught SyntaxError: Unexpected token import
vendor-bundle.js:11197 Uncaught TypeError: Cannot read property '__useDefault' of undefined`

**Comment by zamotany** [comment]
So I've run some tests on **Windows 10** and **Ununtu 16.04** (VM on Windows), both with **Node v6.3.1** and **npm v.3.10.3** with **library v0.17.0**. 

All project were created via `au new` (ESNext).

All tests had same steps:

1. git clone

2. npm install && au run

Here's what I've found:

- If project were created on Windows, it runs on Windows ( both same machine and other) but fails on Ubuntu.

-  If project is created on Ubuntu, it runs on Ubuntu (same machine, could not test on other) and on Windows.

However, in my case errors are slightelly different compared to @jaeko44 results:
```
main.js:1:0:    SyntaxError: import declarations may only appear at top level of a module.
vendor-bundle.js:11197:1    TypeError: target is undefined.
```
Additinally I got 2 warnings:
```
vendor-bundle.js:3482:4    unreachable code after return statement
vendor-bundle.js:11173:45    mutating the [[Prototype]] of an object will cause code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create  
```

Can some owner of Mac:

- Create project on Mac and clone it on Mac, then `npm install && au run`

- Clone [au-test-ubuntu](https://github.com/zamotany/au-test-ubuntu) on Mac and run `npm install && au run`

**Comment by jaeko44** [comment]
@Zamotany 

Are there any temp fixes to get an app running cross-platform? Need to get this running on the server asap.

Previously, I've been able to create a new project using 'au new' and copy over the node_modules folder to the failed project and it would fix the ENONET error I was receiving. (Windows -> Windows)

However, when crossing platforms (Windows -> Ubuntu) I still receive the SyntaxError error after moving node_modules.

**Comment by jaeko44** [comment]
** Temp Fix: (Windows -> Windows) ENOENT Error **

1. Create a new project using 'au new'
2. Copy over the contents of node_modules folder to the old project (make sure to manually install the other dependencies needed in your package.json)

** Temp Fix: (Windows -> Ubuntu / Ubuntu -> Windows) SyntaxError **

Here I might of taken extra steps than necessary, but it worked after:

1. Create a new project using 'au new'
2. Copy over the node_modules folder from the new project.
3. Backup /aurelia_project/aurelia.json (if you have custom dependancies)
4. Delete aurelia_project folder
5. Delete scripts/*-bundle.js
6. Copy over the aurelia_project folder from the new project.
7. Update aurelia.json dependencies to match with your previous aurelia.json file.
8. au run
9. profit ???

**Comment by TimDix** [comment]
Confirmed, I just had the same problem, and the workaround fixed it for me.

**Comment by EisenbergEffect** [comment]
This is a result of you missing the "dot" files like ".babelrc"

**Comment by jaeko44** [comment]
@EisenbergEffect - Could you have it so projects launched with AU RUN have a .gitignore that prevents necessary .files to not be ignored? This will save alot of headaches :)

**Comment by jaeko44** [comment]
@EisenbergEffect This is incorrect, All the dot files are inside the repository.

![ss 2016-08-09 at 12 48 32](https://cloud.githubusercontent.com/assets/9289791/17483877/146f4580-5dcb-11e6-8ff2-598ee2444345.png)

Any idea what else might be causing this?

**Comment by EisenbergEffect** [comment]
Make sure those files are in your filesystem. That is what caused the
problem for me.

**Comment by Mauricio3000** [comment]
**Having same issue**
I created ProjectA via `au run` and confirmed it works. Then I made it a git repo, `git add .`, `git commit -m"initial commit"`using default .gitignore file created by cli.

I then `git clone /home/user/ProjectA  ProjectB`. I cd into ProjectB and try `au run` in ProjectB, but that fails. I try `npm install`, `au run` in ProjectB, but that also fails. (Error output below)

I check number of directories in node_modules for both projects:
`ls -l /home/user/ProjectA/node_modlues | wc -l` Result: 659
`ls -l /home/user/ProjectB/node_modlues | wc -l` Result: 38

I `ls -l /home/user/ProjectA` and `ls -l /home/user/ProjectB` and they have identical files, including "bable.rc"

So npm install failed to install 621 dependencies for ProjectB.

What command do I run, after cloning an aurelia-cli project, to install all dependencies so `au run` works?

**Environment**
MacOS 10.11.15 _running_ Vagrant 1.8.4 _running_ Cent OS 7 3.10.0-327.4.5.el7.x86_64
NodeJs 4.4.7
Npm 2.15.8
Git 1.8.3.1

**ProjectB command/output right after having been cloned:**
`au run`
Only has "New" or "Help" options as if no aurelia-cli project present

`npm install`
npm WARN package.json testingaucli@0.1.0 No README data
npm WARN optional dep failed, continuing fsevents@1.0.14
npm WARN optional dep failed, continuing fsevents@1.0.14
npm WARN optional dep failed, continuing fsevents@1.0.14
npm WARN deprecated tough-cookie@2.2.2: ReDoS vulnerability parsing Set-Cookie https://nodesecurity.io/advisories/130
gulp-rename@1.2.2 node_modules/gulp-rename

babel-plugin-syntax-flow@6.13.0 node_modules/babel-plugin-syntax-flow

gulp-changed-in-place@2.0.3 node_modules/gulp-changed-in-place

karma-jasmine@1.0.2 node_modules/karma-jasmine

connect-history-api-fallback@1.3.0 node_modules/connect-history-api-fallback
...
babel-eslint@6.1.2 node_modules/babel-eslint
├── lodash.assign@4.2.0
├── lodash.pickby@4.6.0
├── babel-types@6.13.0 (to-fast-properties@1.0.2, esutils@2.0.2, lodash@4.15.0, babel-runtime@6.11.6)
├── babylon@6.8.4 (babel-runtime@6.11.6)
└── babel-traverse@6.13.0 (babel-messages@6.8.0, globals@8.18.0, debug@2.2.0, invariant@2.2.1, babel-code-frame@6.11.0, lodash@4.15.0, babel-runtime@6.11.6)

aurelia-cli@0.17.0 node_modules/aurelia-cli
├── aurelia-polyfills@1.0.0 (aurelia-pal@1.0.0)
├── aurelia-dependency-injection@1.0.0 (aurelia-pal@1.0.0, aurelia-metadata@1.0.0)
└── npm@3.10.6

`au run`
Starting 'readProjectConfiguration'...
{ uid: 2,
  name: 'readProjectConfiguration',
  branch: false,
  error: 
   { [Error: ENOENT: no such file or directory, open 'node_modules/aurelia-binding/package.json']
     errno: -2,
     code: 'ENOENT',
     syscall: 'open',
     path: 'node_modules/aurelia-binding/package.json' },
  duration: [ 0, 12114856 ],
  time: 1471294609619 }
{ uid: 1,
  name: '<series>',
  branch: true,
  error: 
   { [Error: ENOENT: no such file or directory, open 'node_modules/aurelia-binding/package.json']
     errno: -2,
     code: 'ENOENT',
     syscall: 'open',
     path: 'node_modules/aurelia-binding/package.json' },
  duration: [ 0, 22763902 ],
  time: 1471294609630 }
{ uid: 0,
  name: '<series>',
  branch: true,
  error: 
   { [Error: ENOENT: no such file or directory, open 'node_modules/aurelia-binding/package.json']
     errno: -2,
     code: 'ENOENT',
     syscall: 'open',
     path: 'node_modules/aurelia-binding/package.json' },
  duration: [ 0, 23912746 ],
  time: 1471294609630 }
{ [Error: ENOENT: no such file or directory, open 'node_modules/aurelia-binding/package.json']
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'node_modules/aurelia-binding/package.json' }

**Comment by Mauricio3000** [comment]
Got some help on the gitter aurelia/Discuss from @BruceBissett @gbpnkans @CasiOo 

npm install -g npm fixed it!

Had to update npm, now using: Npm 3.10.6

**Comment by alexisargyris** [comment]
I'm probably having the same issue. I have an app on a Mac and on Github which builds and runs with no problem using the CLI. I downloaded package.json and /src files on a Win10 machine. Did an 'npm install' and then 'au run --env dev'. I get back this:

```
Tracing resources/elements/notes...
Tracing resources/elements/text...
commonJs.convert: COULD NOT CONVERT: C:/Users/xxx/src/resources/elements/text.js, so skipping it. Error was: Error: Line 81: Unexpected token =
{ uid: 9,
  name: 'writeBundles',
  branch: false,
  error:
   [Error: Parse error using esprima for file: C:/Users/xxx/src/resources/elements/text.js
   Error: Line 81: Unexpected token =],
  duration: [ 5, 324739938 ],
  time: 1473055326244 }
{ uid: 1,
  name: '<series>',
  branch: true,
  error:
   [Error: Parse error using esprima for file: C:/Users/xxx/src/resources/elements/text.js
   Error: Line 81: Unexpected token =],
  duration: [ 6, 943863801 ],
  time: 1473055326269 }
{ uid: 0,
  name: '<series>',
  branch: true,
  error:
   [Error: Parse error using esprima for file: C:/Users/xxx/src/resources/elements/text.js
   Error: Line 81: Unexpected token =],
  duration: [ 6, 959741301 ],
  time: 1473055326282 }
[Error: Parse error using esprima for file: C:/Users/xxx/src/resources/elements/text.js
Error: Line 81: Unexpected token =]
```

I tried removing the file it's complaining about but then it gets stuck on another file. I can't see what's so special about these files because CLI manages to process with no problem other, very similar in structure files (not shown above).

**Comment by jaeko44** [comment]
@alexisargyris 
This is caused due to a bug in the Aurelia-CLI not being able to correctly find the dependancies, in order to fix this replace all '\' in the front/aurelia_project/aurelia.json file into '/'.

**Comment by alexisargyris** [comment]
@jaeko44 
I'm afraid it didn't work:

* I'm on npm 3.10.6,
* did (once again) a 'npm install', 
* replaced in aurelia.json all occurences of '/' (Mac) with '\\\' (Win) 

and I still get exactly the same errors. Am I doing something wrong? Is there any other idea?

Meantime, I'll try next to install in a clean directory to see if that will make any difference.

**Comment by jaeko44** [comment]
Can you try this:

https://github.com/aurelia/cli/issues/253#issuecomment-237270543

**Comment by jaeko44** [comment]
@EisenbergEffect People are still running into issues. The NPM issue was fixed by the latest NPM. However moving from Ubuntu -> Windows, Mac -> Windows or the opposite causes issues. As seen above.

**Comment by wescopeland** [comment]
I can confirm I'm also facing the same or a similar issue. I've created an Aurelia project using the CLI, thrown it onto a Git repo, and when another team members pulls the project down and executes "au run", they're getting the Unexpected token import error.

Cloning the git repo into another directory on my computer also causes the error to occur.

**Comment by zezelazo** [comment]
The solution is to replace all the '\\' for '/', It worked for me on Windows and Mac

**Comment by guser** [comment]
today this error still exists...hmmph. what I am missing. 
main.js:1 Uncaught SyntaxError: Unexpected token import
vendor-bundle.js:11052 Uncaught TypeError: Cannot read property '__useDefault' of undefined

**Comment by AStoker** [comment]
@guser not sure your issue is the same one. From your error it looks like your code isn't being transpiled and bundled (`import` is an ES6 feature, and if a browser hits that when it doesn't support it, it throws an error). You have your vendor bundle there, but it's trying to load up `main.js` which should be inside your `app-bundle.js`

**Comment by guser** [comment]
Thanks for taking time out AStocker. I'm kind of new to this Aurelia. I installed Aurelia-cli fresh. Create a project and did the "au run --watch", it started on Ubuntu 16.04 OK. I have all dependencies installed, but just not working correctly in the browser. I kind of assume the cli would give me a default setup for everything.

**Comment by AStoker** [comment]
Unfortunately, without more details, there are _many_ things that _could_ be going wrong.  
Could you do me a favor, @guser, and write up your question over in Stack Overflow, and then link it here so I can help you there? That way your question doesn't get lost in a closed issue.

**Comment by bpringe** [comment]
I'm having this problem as well. When au new is run, the wizard installs all dependencies including the ones in the aurelia.json file. But later when you don't have the node_modules folder, and you need to reinstall dependencies, you run npm install, but it only installs the dependencies in package.json. This is a small fraction of the packages installed by the au new wizard.

In short, to recreate, create a new cli project with au new and choose to install dependencies. Now delete the node_modules folder. Run npm install. Try to run au run and you will receive an error because several dependencies are missing. 

Maybe the au new wizard should be putting these dependencies in the package.json file when a new project is created? This is the only way for people cloning cli projects to receive all necessary packages via npm install. Am I not understanding something?

**Comment by EisenbergEffect** [comment]
It sounds like you don't have npm 3.x. Can you check that? NPM 3 is required.

**Comment by bpringe** [comment]
Oh I see. So 4.x will not work? I'm using npm 4.0.5. Do I need to downgrade?

**Comment by EisenbergEffect** [comment]
Hmm. Are you sure that's the NPM version? That might be the Node version. They are different. Try running `npm -v`. If you have something earlier than 3.x, then you can run `npm install -g npm` to get the latest.

**Comment by bpringe** [comment]
I got that info from running npm -v. And I also ran npm install -g npm just to be sure.

**Comment by EisenbergEffect** [comment]
Interesting. @jdanyow @niieani Do either of you know anything about this?

**Comment by jdanyow** [comment]
Is it this: https://github.com/aurelia/cli/issues/332

**Comment by bpringe** [comment]
No, it's not a problem with the path separators. Mine has / and works fine. It's just that the dependencies are not all downloaded with npm install after removing the node_modules folder (or pulling from a git repo that ignores node_modules).

**Comment by EisenbergEffect** [comment]
Can you check your npm version with npm -v ?

**Comment by bpringe** [comment]
npm -v yields 4.0.5

**Comment by EisenbergEffect** [comment]
I'm wondering if this is a bug in npm 4.... @jdanyow @niieani Do you guys know anything about this?

**Comment by niieani** [comment]
Not sure what's going on here. Looks like a problem with babel, probably configs not being applied or overridden for some reason.

**Comment by bpringe** [comment]
I found a solution. I'm not sure exactly what the problem was but it had to do with my node version. While I had the latest version of npm, my node version was 4.x. I updated to Node version 6.9.2 and then ran npm install on the cli project and it installed all the dependencies correctly. Thanks for the help.

**Comment by pbgc** [comment]
I'm having the same issue.
- macOS Sierra with node 7.5.0 (installed with brew) and npm 4.1.2 (also tried with npm 4.2.0)

Just tried (several times..) to clone https://github.com/PacktPublishing/Learning-Aurelia
For example in chapter-2/app
npm install
au run

gives me:

unreachable code after return statement[Saber mais] vendor-bundle.js:3558:4
SyntaxError: import declarations may only appear at top level of a module  main.js:1
TypeError: target is undefined[Saber mais]  vendor-bundle.js:11300:1

**Comment by tristan-shelton** [comment]
I just encountered this issue. It seems like it may be an NPM issue rather than an aurelia-cli one.

For me at least switching from Node 8.4.0/NPM 5.3.0 to Node 6.11.3/NPM 3.10.10 solved it and all packages installed correctly.

With NPM 5.3.0 I got `TypeError: Cannot read property '0' of undefined`, seemingly related to https://github.com/npm/npm/issues/17858 but while the installation succeeded with NPM 5.2.0 as suggested in the comments there, the number of packages was still wrong.
